src/BenchmarkDotNet/Diagnosers/DescriptorConfigInjector.cs

```diff
@@ -0,0 +1,16 @@
+﻿using BenchmarkDotNet.Reports;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BenchmarkDotNet.Diagnosers
+{
+    public class DescriptorConfigInjector<TConfig>
```

Comment:
This doesn't look DRY to me. It looks rather pointless. I thought you would create a base metric descriptor that implements the `IMetricDescriptor` interface with virtual method/property implementations.